rr/gstreamer/gst_media.py

```diff
@@ -104,6 +130,31 @@ def stop_media(self):
         if gst.StateChangeReturn.FAILURE == ret:
             raise GstMediaError("Unable to stop the media")
 
+    def install_callback(self, callback):
+        if callback is None:
+            raise GstMediaError("Invalid callback")
+
+        self.callback = callback
+
+    def on_new_buffer(self):
+        try:
+            appsink = self._pipeline.get_by_name("appsink")
+            appsink.connect("new-sample", self._on_new_buffer, appsink)
+            time.sleep(1)
+        except AttributeError as e:
+            raise GstMediaError("Unable to get sample from buffer") from e
```

Comment:
You are not extracting the buffer here, you are just installing the callback. Later, on another thread the appsink will call `self._on_new_buffer` with the buffer. As such this will not emit an exception.

Reply:
Alright, I renamed the exception message this way:

`
raise GstMediaError("Unable to get sample from buffer") from e

raise GstMediaError("Unable to install buffer callback") from e
`